src/internal/SwitchLabel.js

```diff
@@ -0,0 +1,87 @@
+// @flow weak
+/* eslint-disable jsx-a11y/label-has-for */
+
+import React, { Component, PropTypes } from 'react';
+import { createStyleSheet } from 'jss-theme-reactor';
+import classNames from 'classnames';
+
+export const styleSheet = createStyleSheet('SwitchLabel', (theme) => {
+  return {
+    root: {
+      display: 'inline-flex',
+      alignItems: 'center',
+      cursor: 'pointer',
+    },
+    hasLabel: {
+      marginLeft: -12,
+      marginRight: 16, // used for row presentation of radio/checkbox
+    },
+    disabled: {
+      color: theme.palette.text.disabled,
+      cursor: 'not-allowed',
+    },
+  };
+});
+
+export function withSwitchLabel(SwitchComponent) {
+  const switchComponentName = SwitchComponent.displayName || SwitchComponent.name;
+
+  return class SwitchLabel extends Component {
+    static displayName = `withSwitchLabel(${switchComponentName})`;
```

Comment:
Could be using the recompose helper to set the display name:
https://github.com/acdlite/recompose/blob/master/src/packages/recompose/createHelper.js